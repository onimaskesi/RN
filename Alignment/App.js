import React from 'react';
import {SafeAreaView, View, Text, Button, StyleSheet} from 'react-native';

function App() {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.boxes}>
        <View style={styles.box_1} />
        <View style={styles.box_2} />
        <View style={styles.box_3} />
        <View style={styles.box_4} />
      </View>
      <Button
        title="Press Me!"
        onPress={() => {
          console.log('Selam!');
        }}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10,
  },
  boxes: {
    flex: 1,
    backgroundColor: 'yellow',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    margin: 10,
  },
  box_1: {
    height: 75,
    width: 75,
    backgroundColor: 'red',
  },
  box_2: {
    height: 75,
    width: 75,
    backgroundColor: 'blue',
  },
  box_3: {
    height: 75,
    width: 75,
    backgroundColor: 'green',
  },
  box_4: {
    height: 75,
    width: 75,
    backgroundColor: 'orange',
  },
});

export default App;
