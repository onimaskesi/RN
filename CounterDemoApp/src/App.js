import React, {useState} from 'react';
import {SafeAreaView, StyleSheet, Text} from 'react-native';
import Button from './components/Button/Button';

const App = () => {
  const [counter, setCounter] = useState(0);
  const [counterFontSize, setCounterFontSize] = useState(20);

  const addCounter = () => {
    setCounterFontSize(counterFontSize + 1);
    setCounter(counter + 1);
  };

  const styles = getStyles(counterFontSize);

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.title}>Welcome to The Counter App!</Text>
      <Text style={styles.counterStyle}>{counter}</Text>
      <Button title="+" onPress={addCounter} />
    </SafeAreaView>
  );
};

export default App;

const baseStyle = StyleSheet.create({
  text: {
    color: 'white',
    fontSize: 26,
    fontWeight: 'bold',
    margin: 15,
    alignSelf: 'center',
  },
});

const getStyles = counterFontSize =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: 'black',
      justifyContent: 'space-around',
    },
    title: {
      ...baseStyle.text,
    },
    counterStyle: {
      ...baseStyle.text,
      fontSize: counterFontSize,
    },
  });
