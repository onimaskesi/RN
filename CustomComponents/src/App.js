import React from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import Card from './components/Card';

function App() {
  return (
    <SafeAreaView style={styles.container}>
      <Card
        title="Albert Einstein"
        text="Hayal gücü bilgiden daha önemlidir."
      />
      <Card
        title="Charlie Chaplin"
        text="Hayat, uzak çekimde komedi, yakın planda trajedidir."
      />
      <Card title="onimaskesi" text="Hayat bir sabunsa, köpürt beni pakizem." />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#e0e0e0',
  },
});

export default App;
