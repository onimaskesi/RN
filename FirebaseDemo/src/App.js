import React from 'react';
import {SafeAreaView, Text, Button} from 'react-native';
import auth from '@react-native-firebase/auth';
import database from '@react-native-firebase/database';

function App() {
  const signUp = () => {
    auth()
      .createUserWithEmailAndPassword('onimaskesi@gmail.com', 'FramboazlıPasta')
      .then(user => console.log(user))
      .catch(err => console.log(err));
  };

  const signIn = () => {
    auth()
      .signInWithEmailAndPassword('onimaskesi@gmail.com', 'FramboazlıPasta')
      .then(user => console.log(user))
      .catch(err => console.log(err));
  };

  const signOut = () => {
    auth()
      .signOut()
      .then(success => console.log(success))
      .catch(err => console.log(err));
  };

  const checkOut = () => {
    const user = auth().currentUser;
    console.log(user);
  };

  const setDatasToDB = () => {
    database().ref().set({
      name: 'Erdem Onat',
      surname: 'KIRAL',
      nickname: 'onimaskesi',
    });

    const books = [
      {name: 'Rüzgarın Adı', writer: 'Patrick Rothfuss'},
      {name: 'Suç ve Ceza', writer: 'Dostoyevski'},
      {name: 'Kinyas ve Kayra', writer: 'Hakan Günday'},
    ];

    database().ref('/books').set(books);
  };

  const updateFirstBookNameInDB = async () => {
    const firstBookName = 'Rüzgarın Adı';
    const secondBookName = 'Bilge Adamın Korkusu';

    const nameOfTheBook = await database().ref('books/0/name').once('value');
    console.log(nameOfTheBook.val());

    let newName =
      nameOfTheBook.val() === firstBookName ? secondBookName : firstBookName;

    database().ref('/books/0').update({name: newName});
  };

  const pushNewBook = () => {
    database().ref('/books').push({
      name: 'Kavgam',
      writer: 'Adolf Hitler',
    });
  };

  const getAllDatasFromDatabase = () => {
    database()
      .ref()
      .once('value')
      .then(snapshot => {
        const allDatas = snapshot.val();
        console.log(allDatas);
      });
  };

  const getFromDatabase = reference => {
    database()
      .ref(reference)
      .once('value')
      .then(snapshot => {
        const data = snapshot.val();
        console.log(data);
      });
  };

  const checkRealTimeChangesInDB = () => {
    database()
      .ref()
      .on('value', snapshot => {
        const newDatas = snapshot.val();
        console.log(newDatas);
      });
  };

  return (
    <SafeAreaView style={{flex: 1, justifyContent: 'space-between'}}>
      <Text style={{fontSize: 70}}>Hello Firebase</Text>
      <Button title="Sign Up" onPress={signUp} />
      <Button title="Sign In" onPress={signIn} />
      <Button title="Sign Out" onPress={signOut} />
      <Button title="Check The User" onPress={checkOut} />
      <Button title="Set Datas To DB" onPress={setDatasToDB} />
      <Button
        title="Get datas from database"
        onPress={getAllDatasFromDatabase}
      />
      <Button title="Get Books" onPress={() => getFromDatabase('/books')} />
      <Button title="Push New Book" onPress={pushNewBook} />
      <Button
        title="Check Real Time Changes"
        onPress={checkRealTimeChangesInDB}
      />
      <Button
        title="Update name of the first book"
        onPress={updateFirstBookNameInDB}
      />
    </SafeAreaView>
  );
}

export default App;

const user = {
  additionalUserInfo: {isNewUser: true},
  user: {
    displayName: null,
    email: 'onimaskesi@gmail.com',
    emailVerified: false,
    isAnonymous: false,
    metadata: [Object],
    phoneNumber: null,
    photoURL: null,
    providerData: [Array],
    providerId: 'firebase',
    tenantId: null,
    uid: 'a3W682yHF2QHINlzHa9eLRdNbLx1',
  },
};
