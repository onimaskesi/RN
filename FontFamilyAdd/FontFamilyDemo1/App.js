import React from 'react';
import {SafeAreaView, StyleSheet, Text} from 'react-native';

export default () => {
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.textBold}>Gilroy-ExtraBold</Text>
      <Text style={styles.textLight}>Gilroy-Light</Text>
      <Text style={styles.textMosaic}>PaletteMosaic-Regular</Text>
    </SafeAreaView>
  );
};

const baseStyle = StyleSheet.create({
  text: {
    fontSize: 20,
    margin: 10,
  },
});

const {text: baseTextStyle} = baseStyle;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  textBold: {
    ...baseTextStyle,
    fontFamily: 'Gilroy-ExtraBold',
  },
  textLight: {
    ...baseTextStyle,
    fontFamily: 'Gilroy-Light',
  },
  textMosaic: {
    ...baseTextStyle,
    fontFamily: 'PaletteMosaic-Regular',
  },
});
