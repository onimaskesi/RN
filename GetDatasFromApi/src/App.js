import React, {useState, useEffect} from 'react';
import {
  FlatList,
  ActivityIndicator,
  StyleSheet,
  SafeAreaView,
} from 'react-native';
import UserCard from './components/UserCard';
import axios from 'axios';

function App() {
  const [userList, setUserList] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  /*
  async function fetchData() {
    const response = await axios.get(
      'https://jsonplaceholder.typicode.com/users',
    );

    setUserList(response.data);
    setIsLoading(false);
  }
  */

  function fetchData() {
    axios
      .get('https://jsonplaceholder.typicode.com/users')
      .then(response => {
        setUserList(response.data);
        setIsLoading(false);
      })
      .catch(err => console.log(err));
  }

  useEffect(() => {
    fetchData();
  }, []);

  const renderUser = ({item}) => <UserCard user={item} />;

  return (
    <SafeAreaView style={styles.container}>
      {isLoading ? (
        <ActivityIndicator size="large" color="black" />
      ) : (
        <FlatList data={userList} renderItem={renderUser} />
      )}
    </SafeAreaView>
  );
}

export default App;

const styles = StyleSheet.create({
  container: {flex: 1},
});
