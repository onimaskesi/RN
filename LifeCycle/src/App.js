import React, {useState, useEffect} from 'react';
import {SafeAreaView, Text, Switch, View, StyleSheet} from 'react-native';

function App() {
  const [switchValue, setSwitchValue] = useState(false);
  function switchValueChange(value) {
    setSwitchValue(value);
  }
  return (
    <SafeAreaView style={styles.container}>
      <Switch
        style={styles.switch}
        value={switchValue}
        onValueChange={switchValueChange}
      />
      {switchValue && <Hello />}
    </SafeAreaView>
  );
}

export default App;

function Hello() {
  useEffect(() => {
    console.log('Hi bro!');

    return () => {
      console.log('Bye bro!');
    };
  }, []);

  return (
    <View>
      <Text style={styles.helloText}>Hello World!</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 0.5,
    alignItems: 'center',
  },
  switch: {
    margin: 20,
  },
  helloText: {
    backgroundColor: 'black',
    color: 'aqua',
    fontSize: 20,
    fontWeight: 'bold',
    padding: 20,
  },
});
