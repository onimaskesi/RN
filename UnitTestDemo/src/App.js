import React, {useState} from 'react';
import {View, FlatList, TextInput, StyleSheet, Text} from 'react-native';
import Button from '../components/Button';

export default () => {
  const [input, setInput] = useState('');
  const [list, setList] = useState([]);

  function addHHandler() {
    if (input) {
      setList([...list, input]);
      setInput('');
    }
  }

  const renderInput = ({item}) => <Text>{item}</Text>;

  const keyExt = (_, i) => i.toString();

  return (
    <View sytle={styles.container}>
      <FlatList
        testID="list"
        data={list}
        renderItem={renderInput}
        keyExtractor={keyExt}
      />
      <TextInput
        testID="input-area"
        style={styles.input}
        value={input}
        onChangeText={setInput}
      />
      <Button title="Ekle" onClick={addHHandler} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  input: {
    padding: 5,
    margin: 10,
    borderRadius: 10,
    backgroundColor: 'lightgray',
  },
});
