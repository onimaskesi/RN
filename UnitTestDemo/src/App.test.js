import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import App from './App';

test('should match with snapshot', () => {
  const comp = render(<App />);
  expect(comp).toMatchSnapshot();
});

test('should add item to list', () => {
  const comp = render(<App />);

  const list = comp.getByTestId('list');
  const listDataLength = list.props.data.length;

  const changeText = 'test';
  const inputArea = comp.getByTestId('input-area');
  fireEvent.changeText(inputArea, changeText);

  const buttonForAdd = comp.getByTestId('button-touchable');
  fireEvent.press(buttonForAdd);

  expect(list.props.data.length).toBe(listDataLength + 1);
});
