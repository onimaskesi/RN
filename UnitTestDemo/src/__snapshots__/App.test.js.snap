// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  sytle={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "lightgray",
        "borderRadius": 10,
        "margin": 10,
        "padding": 5,
      }
    }
    testID="input-area"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 10,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="button-touchable"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
      testID="button-title"
    >
      Ekle
    </Text>
  </View>
</View>
`;
